src/anbox/config.cpp

```diff
@@ -88,3 +84,24 @@ anbox::SystemConfiguration& anbox::SystemConfiguration::instance() {
   static SystemConfiguration config;
   return config;
 }
+
+anbox::SystemConfiguration::SystemConfiguration() {
+  auto gen_resource_path = [] () -> fs::path {
+    const auto snap_path = utils::get_env_value("SNAP");
+    if (!snap_path.empty()) {
+      return fs::path(snap_path) / "usr/share/anbox";
+    }
+
+    const std::string exe = utils::process_get_exe_path(::getpid());
+    const std::size_t pos = exe.rfind("/bin/anbox");
+    if (pos != std::string::npos) {
+      const std::string leading_path = exe.substr(0, pos);
+      return fs::path(leading_path) / "share/anbox";
+    } else {
+      return "/usr/local/share/anbox";
```

Comment:
I would prefer if we take this, now that we're attempting to do it right, from CMAKE_INSTALL_PREFIX as this can be modified by the user when calling cmake.

Reply:
So user still need to modify cmake when he encountered the failure of locating `loading_screen.png`. 
I always prefer decoupling build/installation/run since it saves my brain... especially for new program - people want things that "just work".

Btw, I was trying to merge the "directory hierarchy" of `snap` and `make install` - personally i'd like to have consistent hierarchy for both mode such that directories like `data` and `resource` are not assumed for different mode. Still under consideration.

Reply:
When a user encountered the error message then something already went wrong before. Where the *.png files are placed is a packaging decision and therefor something which is set at build time. If you install from source this should follow where the *.png files are installed and they're put into CMAKE_INSTALL_PREFIX. See https://github.com/anbox/anbox/blob/master/CMakeLists.txt#L121

So when something invokes `cmake -DCMAKE_INSTALL_PREFIX=/opt` the code should follow that decision.

Reply:
so, you are suggesting passing cmake configurations to the code and generating the "default path" at build time?

Reply:
Correct. This should only affect `return "/usr/local/share/anbox";` and we already have a config.h.in in src/anbox which is filled by cmake at configuration time.

Reply:
this `config.h.in` is not public yet? i didn't see any when build from this master...